sources/file_table.cpp

```diff
@@ -191,6 +202,20 @@ FileBase* FileTable::open_as(const id_type& id, int type)
         }
     }
 
+    if(!free_pool.done()) {
+        bool being_closed = false;
+        do {
+            being_closed = false;
+            {
+                std::lock_guard<std::mutex> l(m_closing_lock);
+                being_closed = (m_closing_ids.count(id) != 0);
+            }
+            if(being_closed) {
+                free_pool.wait();
+            }
+        } while(being_closed);
+    }
+
```

Comment:
What is the purpose of this block of code?

Reply:
This is to make sure that the file is not currently being deleted by one of the background threads -- I don't know the FileBase implementation well enough to know that it would be safe to have two FileBases referring to the same file open, with one actively being deleted by a background thread. 

In pseudocode, it roughly does:

    while(the file I'm trying to open is being closed) {
        wait for at least one background thread to finish (which would indicate a change to the list of files being closed)
    }